include/deal.II/base/polynomial.h

```diff
@@ -585,6 +586,119 @@ namespace Polynomials
+   * the derivative degree of freedom in $x=0$ and $x=1$, respectively. As
+   * such, they are zero in both the end points $x=0, x=1$ and have zero
+   * derivative in the opposite ends, $p_1'(1)=0$ and $p_{n-1}'(0)=0$. As
+   * opposed to the original Hermite polynomials, $p_0$ does not have zero
+   * derivative in $x=0$. The additional degree of freedom is used to make
+   * $p_0$ and $p_1$ orthogonal, which results in a root at $2/7$ for
+   * $n=3$. Furthermore, the extension of these polynomials to higher degrees
+   * $n>3$ is constructed by adding additional nodes inside the unit interval,
+   * again ensuring better conditioning. The nodes are computed as the roots
+   * of the Jacobi polynomials for $\alpha=\beta=2$ which are orthogonal
+   * against the generating function $x^2(1-x)^2$ of the Hermite
+   * functions. For example at $n=4$, all of $p_0, p_1, p_3, p_4$ get an
+   * additional root at $x=0.5$ through the factor $(x-0.5)/x_0$.
+
+   * These two relaxations improve the condition number of the mass matrix
+   * (i.e., interpolation) significantly, as can be seen from the following
+   * table:
+   *
+   * <table align="center" border="1">
+   *   <tr>
+   *    <th>&nbsp;</th>
+   *    <th colspan="2">Condition number mass matrix</th>
+   *   </tr>
+   *   <tr>
+   *    <th>degree</th>
+   *    <th>HermiteInterpolation</th>
+   *    <th>HermiteLikeInterpolation</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=3</th>
+   *    <th>1057</th>
+   *    <th>21.40</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=4</th>
+   *    <th>6580</th>
+   *    <th>15.52</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=5</th>
+   *    <th>1.875e+04</th>
+   *    <th>18.52</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=6</th>
+   *    <th>6.033e+04</th>
+   *    <th>19.42</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=10</th>
+   *    <th>9.756e+05</th>
+   *    <th>27.85</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=15</th>
+   *    <th>9.431e+06</th>
+   *    <th>40.48</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=25</th>
+   *    <th>2.220e+08</th>
+   *    <th>68.30</th>
+   *   </tr>
+   *   <tr>
+   *    <th>n=35</th>
+   *    <th>2.109e+09</th>
+   *    <th>98.06</th>
+   *   </tr>
+   * </table>
+   *
+   * This polynomial inherits the advantageous property of Hermite polynomials
+   * where only two functions have value and/or derivative nonzero on a face
+   * but gives better condition numbers of interpolation, which improves the
+   * performance of some iterative schemes like conjugate gradients with
+   * point-Jacobi.
+   *
+   * @note This class requires LAPACK support.
+   *
+   * @author Martin Kronbichler
+   * @date 2018
+   */
+  class HermiteLikeInterpolation : public Polynomial<double>
+  {
+  public:
+    /**
+     * Constructor for polynomial with index <tt>p</tt> of degree @p
+     * degree_basis.
+     */
+    HermiteLikeInterpolation (const unsigned int degree_basis,
+                              const unsigned int p);
```

Comment:
I didn't see this discussed in the introduction: What is the difference between degree and index? (Also: `p` usually denotes the *polynomial degree*, so if that's not what it is here, maybe we should have a different name for the variable?)

Reply:
That's a good idea, let's call the variable `index`. I agree that we use `p` for the degree, but I still wanted to spell it out in the variable name.